src/store.js

```diff
@@ -488,17 +501,53 @@ const piles = (previousState = {}, action) => {
 
       depilePiles.forEach(pile => {
         pile.items.forEach(itemId => {
+          const label = newState[itemId].label.length
+            ? [newState[itemId].label[0]]
+            : [];
+
           newState[itemId] = {
             ...newState[itemId],
             items: [itemId],
             x: pile.x,
-            y: pile.y
+            y: pile.y,
+            label
           };
         });
       });
       return newState;
     }
 
+    case 'SET_LABEL': {
```

Comment:
Let's create a simple getter/setter property defining the name of the label property. Sorry I didn't specify this earlier but it'll simplify things quite a bit.